doc/buip-hf.md

```diff
@@ -0,0 +1,103 @@
+BUIP-HF Development Notes
```

Comment:
Most of this document was applicable to working on my personal repo.

I think this document does not need to remain, or can be trimmed down do bare essentials as the work continues on BU's `dev` branch.